src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0/Web3.Unity Samples/Scripts/Scenes/MetaMaskLoginProvider.cs

```diff
@@ -17,12 +19,39 @@
 public class MetaMaskLoginProvider : LoginProvider, IWeb3BuilderServiceAdapter
 {
     [SerializeField] private Button loginButton;
+    #if UNITY_WEBGL && !UNITY_EDITOR
+    private MetaMaskController metaMaskController;
+    #endif
 
     protected override void Initialize()
     {
         base.Initialize();
-
         loginButton.onClick.AddListener(LoginClicked);
+        #if UNITY_WEBGL && !UNITY_EDITOR
+        metaMaskController = Object.FindObjectOfType<MetaMaskController>();
+        #endif
+    }
+    
+    public override async Task TryLogin()
+    {
+        try
+        {
+            #if UNITY_WEBGL && !UNITY_EDITOR
+            metaMaskController.ConnectedTsc = new TaskCompletionSource<string>();
+            #endif
+            await base.TryLogin();
```

Comment:
This will call `MetamaskController.Connect()` which will set `  ConnectedTsc = new TaskCompletionSource<string>();` so it's redundant